src/main/java/christmas/domain/promotion/discount/DiscountContext.java

```diff
@@ -0,0 +1,53 @@
+package christmas.domain.promotion.discount;
+
+import christmas.domain.consumer.Orders;
+import christmas.domain.consumer.VisitDay;
+import christmas.domain.promotion.badge.BadgePromotion;
+
+import java.util.Arrays;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class DiscountContext {
+    private final List<DiscountPromotion> applicablePromotions;
+
+    //== Constructor ==//
+    private DiscountContext(
+            VisitDay visitDay,
+            Orders orders,
+            BadgePromotion badge
+    ) {
+        this.applicablePromotions = Arrays.stream(DiscountPromotion.values())
+                .filter(promotion -> promotion.isPromotionPeriod(visitDay))
+                .filter(promotion -> promotion.isApplicable(visitDay, orders, badge))
+                .toList();
+    }
+
+    //== Static Factory Method ==//
+    public static DiscountContext create(
+            VisitDay visitDay,
+            Orders orders,
+            BadgePromotion badge
+    ) {
+        return new DiscountContext(visitDay, orders, badge);
+    }
+
+    //== Utility Method ==//
+    public EnumMap<DiscountPromotion, Integer> applyPromotions(
+            VisitDay visitDay,
+            Orders orders
+    ) {
```

Comment:
🤔**고민**🤔

SonarLint 에 따르면, 구체적인 구현체인 `EnumMap`를 반환하는 것 보단 인터페이스인 `Map` 을 사용하는 걸 권장한다고 합니다. `EnumMap` 을 반환하고 외부에서 계속 쓰는 것 보다 `Map` 을 사용하는 건 어떨까요? 구현체를 외부로 노출해선 안된다고 주장합니다.

```
Java Collections API는 수집 구현 세부 정보를 숨기도록 설계된 인터페이스의 잘 구성된 계층 구조를 제공합니다.
목록, 집합, 지도 등 다양한 수집 데이터 구조의 경우 특정 인터페이스(java.util.List, java.util.Set, java.util.Map)가 필수 기능을 포함합니다.
컬렉션을 메서드 매개 변수, 반환 값으로 전달하거나 필드를 노출할 때는 구현 클래스 대신 이러한 인터페이스를 사용하는 것이 일반적으로 권장됩니다.
java.util.LinkedList, java.util.ArrayList 및 java.util.HasMap과 같은 구현 클래스는 컬렉션 인스턴스화에만 사용해야 합니다.
이러한 구조의 성능 특성을 보다 세밀하게 제어할 수 있으며 개발자는 사용 사례에 따라 선택합니다.
예를 들어 빠른 랜덤 요소 접근이 필수적이라면 java.util.ArrayList를 인스턴스화해야 합니다.
만약 임의의 위치에 요소를 삽입하는 것이 중요하다면 java.util.LinkedList를 선호해야 합니다.
하지만 이것은 API가 노출해서는 안 되는 구현 정보입니다.
```

```suggestion
    public Map<DiscountPromotion, Integer> applyPromotions(
            VisitDay visitDay,
            Orders orders
    ) {
```